oak_functions_service/src/lib.rs

```diff
@@ -145,15 +145,53 @@ impl schema::OakFunctions for OakFunctionsService {
 
     fn update_lookup_data(
         &mut self,
-        lookup_data: &schema::LookupData,
-    ) -> Result<schema::Empty, micro_rpc::Status> {
-        let data = lookup_data
+        request: &schema::UpdateLookupDataRequest,
+    ) -> Result<schema::UpdateLookupDataResponse, micro_rpc::Status> {
+        let update_action = map_update_action(request.action)?;
+
+        let data = request
+            .chunk
+            .as_ref()
+            .unwrap()
             .items
             .iter()
-            .map(|entry| Ok((entry.key.clone(), entry.value.clone())))
-            .collect::<Result<_, micro_rpc::Status>>()?;
+            .map(|entry| (entry.key.clone(), entry.value.clone()))
+            .collect();
+
+        let update_status = self.lookup_data_manager.update_data(update_action, data);
+
+        Ok(schema::UpdateLookupDataResponse {
+            update_status: map_update_status(update_status).into(),
+        })
+    }
+}
+
+// This crate depends on oak_functions_lookup and schema, so we cannot use schema in
+// oak_functions_lookup. As a work-around we define and map a corresponding type for UpdateAction.
+fn map_update_action(action: i32) -> Result<oak_functions_lookup::UpdateAction, micro_rpc::Status> {
```

Comment:
Why does this take an i32 and parses it by itself while the one below takes an already parsed enum value?

Reply:
Changed it to now parse i32 in main body.